### PR TITLE
Migrate allow_list_dropdown_count_users.php into a real controller

### PR DESCRIPTION
**Pull request by AdrienClairembault**
Migrate `ajax/form/allow_list_dropdown_count_users.php` into a real controller.
I've also fixed a few bugs (broken selector, bad count) and updated the tests to make sur it doesn't break again.

| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets |

### CODE REVIEW

File: phpunit/functional/Glpi/Form/AccessControl/ControlType/AllowListDropdownTest.php [hunk 1]
```diff
@@ -44,11 +45,19 @@
...
+            expected_users_count: 7,
+        );
+    }
+
     public function testEmptyAllowList(): void
     {
         $this->checkCountUserForCriteria(
             criteria: [],
-            expected_users_count: countElementsInTable(User::getTable()),
+            expected_users_count: 0,
```
Comment (AdrienClairembault) [inline phpunit/functional/Glpi/Form/AccessControl/ControlType/AllowListDropdownTest.php:60]: I've changed the behavior, an empty list should not allow any user.

File: src/Glpi/Form/AccessControl/ControlType/AllowListDropdown.php [hunk 2]
```diff
@@ -63,6 +65,86 @@ public static function countUsersForCriteria(
...
+    protected static function getTypes(): array
+    {
+        return [
+            User::getType(),
+            Profile::getType(),
+            Group::getType(),
+        ];
+    }
+
+    protected static function countUsers(
```
Comment (AdrienClairembault) [inline src/Glpi/Form/AccessControl/ControlType/AllowListDropdown.php:93]: New method to count users more efficiently, the previous one used the search engine which take the current entities into account.
We want an accurate number here no matter what the current entity of the user.

File: templates/pages/admin/form/access_control/allow_list.html.twig [hunk 3]
```diff
@@ -56,18 +56,20 @@
 
 <script>
     // Each time the dropdown is modified, populate the helper link
-    $("select[name='_allow_list_dropdown[]']").on('change', async function() {
-        const response = await $.post(
-            CFG_GLPI.root_doc + '/ajax/form/allow_list_dropdown_count_users.php',
+    const $select = $("select[name='_access_control[{{ access_control.getId() }}][_allow_list_dropdown][]']")
```
Comment (AdrienClairembault) [inline templates/pages/admin/form/access_control/allow_list.html.twig:59]: Selector was broken after some changes done by another PR.
Added an e2e test to make sure it doesn't break again.

**Comment by AdrienClairembault** [comment]
Rebased to latest main.

**Comment by AdrienClairembault** [comment]
Rebased to latest main (PHPStan level 4 was checked)

**Review by cedric-anne — APPROVED** [review]
